source/request_response.c

```diff
@@ -959,8 +979,28 @@ struct aws_http_message *aws_http2_message_new_from_http1(
         /* append lower case name to the buffer */
         aws_byte_buf_append_with_lookup(&lower_name_buf, &header_iter.name, aws_lookup_table_to_lower_get());
         struct aws_byte_cursor lower_name_cursor = aws_byte_cursor_from_buf(&lower_name_buf);
+        enum aws_http_header_name name_enum = aws_http_lowercase_str_to_header_name(lower_name_cursor);
+        switch (name_enum) {
+            case AWS_HTTP_HEADER_TRANSFER_ENCODING:
+            case AWS_HTTP_HEADER_UPGRADE:
+            case AWS_HTTP_HEADER_KEEP_ALIVE:
+            case AWS_HTTP_HEADER_PROXY_CONNECTION:
+                /**
+                 * An intermediary transforming an HTTP/1.x message to HTTP/2 MUST remove connection-specific header
+                 * fields as discussed in Section 7.6.1 of [HTTP]. (RFC=9113 8.2.2)
+                 */
+                AWS_LOGF_TRACE(
+                    AWS_LS_HTTP_GENERAL,
+                    "Skip connection-specific headers - \"%.*s\" ",
+                    (int)lower_name_cursor.len,
+                    lower_name_cursor.ptr);
+                aws_byte_buf_reset(&lower_name_buf, false);
+                continue; /* jump to next for-loop iteration */
```

Comment:
this seems fragile: repeating the `aws_byte_buf_reset()` from the end of the for-loop.

how about moving `aws_byte_buf_reset()` to the start of the for-loop instead

or having a `bool copy_header = true;` and then you do this:
```
copy_header = false;
break;
```
instead of the weird continue-from-case